[Magiclysm] adds lake start and new retreat island

curstwist opened the pull request:
#### Summary
 ```SUMMARY: Mods "[Magiclysm] adds a new scenario start on an island retreat, the Wizard's Vacation."```


#### Purpose of change
I wanted a magiclsym lake start.  This allows for novice magic users to start on a secluded island retreat.  

#### Describe the solution
Add island: includes a cozy tower and boat house.  Largely wooded.
Add start scenario: novice professions allowed, cost is 6 points due to the locations seclusion, relative safety, and the solid chance for multiple magic loots.

I've added new distribution item groups and incorporated some of the SUS principles for collections.  These will need to be refined with player feedback.  I've also included a great deal of the new magic items and made sure scrolls start spawning more in general.

#### Testing
Lots of in world testing for spawns and mapgen.

#### Additional context
The spoiler pictures!
As with most magic buildings, this is hidden on the overmap but I expect the island shape will give it away.  I may re purpose the lake shape for some vanilla red herrings.

overmap
![overmap](https://user-images.githubusercontent.com/39442864/67308419-12597580-f4c8-11e9-8de8-d9cd3778a78f.PNG)

boathouse:
![boathouse](https://user-images.githubusercontent.com/39442864/67308428-17b6c000-f4c8-11e9-9fa8-01c1ac2de2cb.PNG)

tower levels ground to z4:
![retreat ground](https://user-images.githubusercontent.com/39442864/67308456-2309eb80-f4c8-11e9-99a1-a5f611848906.PNG)
![retreat z1](https://user-images.githubusercontent.com/39442864/67308462-269d7280-f4c8-11e9-85fa-df732108b367.PNG)
![retreat z2](https://user-images.githubusercontent.com/39442864/67308463-269d7280-f4c8-11e9-9669-c79b1cefd81c.PNG)
![retreat z3](https://user-images.githubusercontent.com/39442864/67308473-27ce9f80-f4c8-11e9-9bba-f4d81f97183d.PNG)
![retreat z4](https://user-images.githubusercontent.com/39442864/67308480-29986300-f4c8-11e9-9ff6-f7dd316e6d85.PNG)

KorGgenT commented on data/mods/Magiclysm/itemgroups/itemgroups.json:
```diff
@@ -101,9 +101,12 @@
-      { "group": "spellbook_tier_1", "prob": 50 },
-      { "group": "spellbook_tier_2", "prob": 35 },
-      { "group": "spellbook_tier_3", "prob": 20 },
+      { "group": "spellbook_tier_1", "prob": 40 },
```
these should be probably combined to use a `spellbook_loot_` tier (found in magiclysm/itemgroups/spellbooks.json)

curstwist commented:
> 
> 
> So, for the item groups: I'd like them to be separated out a little more. there should be places where you can find +1 items, but _never_ a +2 item for example. many of the locations that have little to no fighting required in order to obtain the items should be a lower power level in general. This mage's tower might qualify for an exception, but it should be almost orders of magnitude rarer to find a +3 than a +2 intelligence ring for example, which might be easier if separated out into an itemgroup, and then a custom itemgroup for the tower include these item subgroups.
> 
> The most obvious item groups should be something like
> rings -> common, uncommon, rare, etc. you can pretty much follow the material they're made of, aptronym and i talked about rarity at length and they're roughly in that order.
> 
> wands -> minor, lesser, greater.
> This is actually only a rough selection, as spells have their own "tier" which i've actually designated in a more "unofficial" way in the spellbooks.json in itemgroups, you can look at the spell scrolls. This would make wands have maybe 6 or so tiers off the bat.
> 
> melee weapons -> +1, +2
> 
> additionally, the other distributions could use a bit more description in the comment about the expected rarity of the items contained within.

reworked item groups into strength tiers and then incorporated them into the distribution groups.  I think they came out well from my tests.  May need to nerf wand drops, but I think they are fine for now.  I got at the most 2 minor wands.  I removed all higher tier loots except for incredibly rare (1% chance) drops in the small items distribution group.
